src/comm.F

```diff
@@ -8585,57 +6002,71 @@ subroutine getcorner3_2d(s)
 
       if((ibe.eq.0.or.ibe.eq.2) .and. (ibs.eq.0.or.ibs.eq.2))then
         nr = nr + 1
+        !$acc host_data use_device(se2) if(gdirect)
         call mpi_irecv(se2,cmp*cmp,MPI_REAL,myse,tag4,MPI_COMM_WORLD,   &
                        reqs(nr),ierr)
+        !$acc end host_data
         index_se = nr
       endif
 
 !------------------------------------------------------------------
 
       if((ibe.eq.0.or.ibe.eq.2) .and. (ibs.eq.0.or.ibs.eq.2))then
+        !$acc parallel loop gang vector default(present) if(openacc)
```

Comment:
do we need `collapse(2)` clause for all the calculations below?

Reply:
Good catch just fixed it.